### ISSUE TITLE
Spacy Sklearn Interpreter every time load nlp

### ISSUE THREAD

**Issue by AndriiKr** [golastmile/rasa_nlu#190]
Hello!

I found that `interpreter = SpacySklearnInterpreter(**metadata)` every time load heavy spaCy models `nlp = spacy.load(language_name, parser=False, entity=False, matcher=False)`. It's not a problem if we are using rasa NLU as a HTTP server because nlp calling once in the start. But if we using rasa_nlu classes directly from python we need call it  every time.
Here is my snippet:
```
i = 0
bot_ids = []
for entity in db.entities.find():
    for intent in db.intents.find():
        bot_ids.append(entity['bot_id'])
        bot_ids.append(intent['bot_id'])
        i = i + 1
botIdList = list(set(bot_ids))
for bot_id in botIdList:
    print (bot_id)
    metadata = json.loads(open('models/' + bot_id + '/metadata.json').read())
    interpreter = SpacySklearnInterpreter(**metadata)
```
I'm planning that my code will use hundreds and thousands of bots. I can't  to call nlp for every bot better to call it once in the start of the code.

We need possibility add nlp as argument. I think it's could solve the problem. 

Something like this:
```
nlp = spacy.load('en')
interpreter = SpacySklearnInterpreter(metadata, nlp)
```

**Comment by tmbo** [comment]
This might be fixed with the multitenancy pr #85 

But why would you create an interpreter for every message instead of reusing it?

**Comment by jplorandi** [comment]
Because he is trying to avoid keeping each model in RAM. Reusing the interpreter makes sense if you can hold in RAM enough of them; Otherwise, you would like to:
* Load the Word Vectors once (on MITIE this is the `total_word_feature_extractor`)
* Just-In-Time load the models when instancing the interpreter
* Serve a single query
* Destroy the interpreter (`del interpreter`)

This of course means that each call will incurr the overhead of loading the model **every time** thus increasing the CPU costs per served call, but would allow to serve a large number of requests with somewhat limited RAM (Say, if every model weights 100MB or less, you could fit 150 models in RAM in a 16 GB machine, thus being able to serve 150 concurrent requests)

This is a way to handle multitenancy by sacrificing latency. In a dedicated server, you do the inverse: you don't handle multitenancy (or at least you don't do so in large numbers), but you get the best performance out of rasa_nlu.

I'm very interested in this, and as an interim solution for my use case, which is very similar, I'm going to start a docker instance per request (even if answering each takes over 5 seconds).

**Comment by tmbo** [comment]
So yeah this use case should be handled after #85 is merged. One backend is than shared with all its models.

**Comment by AndriiKr** [comment]
I  decided  to use option 3 spaCy + sklearn + MITIE it's much better, faster and lighter then others. I like how training do: 
```
    training_data = TrainingData(path, 'mitie_sklearn', 'en')
    cpuQ = multiprocessing.cpu_count()
    extractor = "./MITIE-models/english/total_word_feature_extractor.dat"
    trainer = MITIESklearnTrainer(extractor, 'en', cpuQ)
    trainer.train(training_data)
```
I can use extractor outside once but while parsing it's impossible againe.
It would be cool if we could use extractor separate from metadata
like this:
```
metadata = json.loads(open('/path/to/metadata.json').read())
extractor = "./MITIE-models/english/total_word_feature_extractor.dat"
interpreter = SpacySklearnInterpreter(metadata, extractor)
```

**Comment by tmbo** [comment]
This should be possible once the multitenancy is merged (PR #85 ).
